### PR TITLE
HPCC-28624 Display additional details of indexes on detail page

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <Jeremy.Clements@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jeclrsg** [comment]
@GordonSmith NPM build step failed because of some missing properties in the WsDFU service in comms (https://github.com/hpcc-systems/Visualization/pull/4055). Will have to update the package files in here w/ the new version number.

**Comment by GordonSmith** [comment]
@jeclrsg I will review once the upstream changes are available...

**Comment by ghalliday** [comment]
@GordonSmith is this now possible to review?

File: esp/src/src-react/components/IndexFileSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    const [DeleteConfirm, setShowDeleteConfirm] = useConfirm({
+        title: nlsHPCC.Delete,
+        message: nlsHPCC.YouAreAboutToDeleteThisFile,
+        onSubmit: React.useCallback(() => {
+            dfuService.DFUArrayAction({ Type: WsDfu.DFUArrayActions.Delete, LogicalFiles: { Item: [file.Name] } }).then(({ ActionResults }) => {
+                const actionInfo = ActionResults?.DFUActionInfo;
+                if (actionInfo && actionInfo.length && !actionInfo[0].Failed) {
+                    replaceUrl("/files");
+                }
+            });
```
Comment (GordonSmith) [inline esp/src/src-react/components/IndexFileSummary.tsx:57]: Error handling

File: esp/src/src-react/components/IndexFileSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        { key: "divider_1", itemType: ContextualMenuItemType.Divider, onRender: () => <ShortVerticalDivider /> },
+        {
+            key: "save", text: nlsHPCC.Save, iconProps: { iconName: "Save" }, disabled: !canSave,
+            onClick: () => {
+                file?.update({
+                    UpdateDescription: true,
+                    FileDesc: description,
+                    Protect: _protected ? WsDfu.DFUChangeProtection.Protect : WsDfu.DFUChangeProtection.Unprotect,
+                    Restrict: restricted ? WsDfu.DFUChangeRestriction.Restrict : WsDfu.DFUChangeRestriction.Unrestricted,
+                });
```
Comment (GordonSmith) [inline esp/src/src-react/components/IndexFileSummary.tsx:113]: Error handling

File: esp/src/src/Utility.ts [hunk 3]
```diff
@@ -111,7 +111,7 @@ export function convertedSize(intsize: number): string {
     if (intsize === null || intsize === undefined) {
         return "";
     } else {
-        const x = Math.floor(Math.log(intsize) / Math.log(1024));
+        const x = intsize > 0 ? Math.floor(Math.log(intsize) / Math.log(1024)) : 0;
```
Comment (GordonSmith) [inline esp/src/src/Utility.ts:114]: Should the `if` above just be a `isNaN`?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Review by GordonSmith — APPROVED** [review]

**Comment by GordonSmith** [comment]
@ghalliday this has been approved.

**Comment by ghalliday** [comment]
I will merge this after the next set of rcs

**Review by ghalliday — APPROVED** [review]

**Pull request merged**